lib/private/Files/Type/Loader.php

```diff
@@ -109,31 +111,38 @@ public function reset() {
 	 * Store a mimetype in the DB
 	 *
 	 * @param string $mimetype
-	 * @param int inserted ID
+	 * @return int inserted ID
 	 */
 	protected function store($mimetype) {
-		$this->dbConnection->insertIfNotExist('*PREFIX*mimetypes', [
-			'mimetype' => $mimetype
-		]);
-
-		$fetch = $this->dbConnection->getQueryBuilder();
-		$fetch->select('id')
-			->from('mimetypes')
-			->where(
-				$fetch->expr()->eq('mimetype', $fetch->createNamedParameter($mimetype)
-			));
-
-		$result = $fetch->execute();
-		$row = $result->fetch();
-		$result->closeCursor();
+		$row = $this->atomic(function () use ($mimetype) {
+			$insert = $this->dbConnection->getQueryBuilder();
+			$insert->insert('mimetypes')
+				->values([
+					'mimetype' => $insert->createNamedParameter($mimetype)
+				])
+				->executeStatement();
```

Comment:
To avoid parallel errors:
```suggestion
				]);
			try {
				$insert->executeStatement();
			} catch (DbalException $e) {
				if ($e->getReason() !== DbalException::REASON_UNIQUE_CONSTRAINT_VIOLATION) {
					throw $e;
				}
			}
```

But there is no need to read when we just wrote it, so could even go with:
```suggestion
				]);
			try {
				$insert->executeStatement();
				return [
					'mimetype' => mimetype,
					'id' => $insert->getLastInsertId(),
				];
			} catch (DbalException $e) {
				if ($e->getReason() !== DbalException::REASON_UNIQUE_CONSTRAINT_VIOLATION) {
					throw $e;
				}
			}
```